src/charm.py

```diff
@@ -54,8 +55,22 @@ def _on_dns_record_relation_changed(self, event: ops.RelationChangedEvent) -> No
         self.dns_record.update_relation_data(relation, relation_provider_data)
         self.unit.status = ops.ActiveStatus()
 
+    def _on_collect_status(self, event: ops.CollectStatusEvent) -> None:
+        """Handle collect status event.
+
+        Args:
+            event: Event triggering the collect-status hook
+        """
+        try:
+            relation_requirer_data = self.dns_record.get_remote_relation_data()
+        except ValueError as err:
+            logger.info("Validation error of the relation data: %s", err)
+            event.add_status(ops.BlockedStatus(f"Validation error of the relation data: {err}"))
+            return
+        self.bind.collect_status(event, relation_requirer_data)
```

Comment:
I'm wondering if it might be worth separating the charm context from bind module so that we don't couple them together.
I'm thinking perhaps `self.bind.check_conflicts` and then setting a `BlockedStatus` from the `charm.py` :)

Reply:
Yes but I was thinking that each module could have its collect_status() that would be called by the _on_collect_status() in the charm. In that sense it would be more something akin to the module than the charm itself.  
I'm not completely sure.